samples/bootloader/README.rst

```diff
@@ -81,34 +79,128 @@ Provisioning
+   Therefore, the bootloader will refuse to boot if any hash contains a half-word with the value ``0xFFFF``.
+   If your public key hash is found to contain this value, please :ref:`regenerate it <ug_fw_update_keys>` or use another public key.
+
+#. Writing data to an OTP region means that provisioned data cannot be written more than once to the target device.
+   When programming images that contain flash memory content for this region, such as the bootloader or images containing the bootloader, the UICR must first be erased.
+
 .. note::
-   On some SoCs/SiPs, like the nRF5340 or the nRF9160, the provisioned data is held in the *One-time programmable* (OTP) region in the *User Information Configuration Registers* (UICR).
-   For this reason, please note the following:
+   On the nRF9160, the UICR can only be erased by erasing the whole chip.
+
+To perform a whole chip erase, do the following:
+
+.. tabs::
 
-   * You must erase the UICR before programming the bootloader.
-     On the nRF9160, the UICR can only be erased by erasing the entire chip:
+   .. tab:: Command Line
 
-     * On the command line, call ``west flash`` with the ``--erase`` option to erase the chip, like in the following example:
+      Using ``west``:
 
-     .. parsed-literal::
-        :class: highlight
+      .. code-block:: console
 
         west flash -d *build_directory* --erase
 
-       This will erase the whole chip before programming the new image.
+      Using ``nrfjprog``:
+
+      .. code-block:: console
+
+         nrfjprog -f NRF91 --eraseall
+
+   .. tab:: |SES|
+
+      Choose :guilabel:`Target` -> :guilabel:`Connect J-Link` and then :guilabel:`Target` -> :guilabel:`Erase All`.
+
+.. _bootloader_flash_layout:
+
+Flash memory layout
+===================
+
+The flash memory layout is defined by the :file:`samples/bootloader/pm.yml` file, which establishes four main partitions:
+
+1. *B0* - Contains the bootloader sample.
+#. *Provision* - Stores the provisioned data.
+#. *S0* - Defines one of the two potential storage areas for the second stage bootloader.
+#. *S1* - Defines the other one of the two potential storage areas for the second stage bootloader.
+
+The default location for placing the next image in the boot chain is *S0*.
+This would result, for example, in a flash memory layout like the following, when using the ``nrf52840dk_nrf52840`` board:
+
+.. figure:: ../../doc/nrf/images/b0_flash_layout.svg
+   :alt: B0 flash memory layout
+
+   B0 flash memory layout
 
-     * In |SES|, choose :guilabel:`Target` -> :guilabel:`Connect J-Link` and then :guilabel:`Target` -> :guilabel:`Erase All` to erase the whole chip.
+When the application utilizes a second-stage upgradable bootloader, *S1* can be simultaneously programmed with the same content as *S0* when compiled with :ref:`upgradable_bootloader_presigned_variants`.
```

Comment:
Same content, or the same codebase compiled for S1?

Reply:
Changed in latest commit as part of addressing [this comment above](https://github.com/nrfconnect/sdk-nrf/pull/3607#discussion_r611463068).